inject.c

```diff
@@ -1,38 +1,60 @@
 #include <frida-core.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/xattr.h>
 
 int
 main (int argc, char * argv[])
 {
   int result = 0;
+  const char * path = "/data/local/tmp/android-inject-example/agent.so";
+  const char * context = "u:object_r:frida_file:s0";
   FridaInjector * injector;
   int pid;
   GError * error;
   guint id;
 
+  frida_init ();
+
   if (argc != 2)
     goto bad_usage;
 
   pid = atoi (argv[1]);
   if (pid <= 0)
     goto bad_usage;
 
-  frida_init ();
+  frida_selinux_patch_policy ();
+
+  if (setxattr (path, XATTR_NAME_SELINUX, context, strlen (context) + 1, 0) != 0)
+  {
+    g_printerr ("Failed to set SELinux permissions\n");
+    goto bad_init;
+  }
 
   injector = frida_injector_new ();
 
   error = NULL;
-  id = frida_injector_inject_library_file_sync (injector, pid, "./agent.so", "example_agent_main", "example data", &error);
+  id = frida_injector_inject_library_file_sync (injector, pid, path, "example_agent_main", "example data", NULL, &error);
   if (error != NULL)
   {
-    fprintf (stderr, "%s\n", error->message);
+    g_printerr ("%s\n", error->message);
+    g_error_free (error);
+
+    result = 1;
+  }
+
+  error = NULL;
+  frida_injector_close_sync (injector, NULL, &error);
+  if (error != NULL)
+  {
+    g_printerr ("%s\n", error->message);
     g_error_free (error);
 
     result = 1;
   }
```

Comment:
```suggestion
  frida_injector_close_sync (injector, NULL, NULL);
```

We can drop the error-handling here and just pass `NULL`, as this can only fail if a `Cancellable` is passed and it gets cancelled.